v2/internal/loader/counting_loader.go

```diff
@@ -9,13 +9,13 @@ import (
 	"github.com/multiformats/go-varint"
 )
 
-// counter tracks how much data has been read.
-type counter struct {
-	totalRead uint64
+// Counter tracks how much data has been read.
+type Counter struct {
```

Comment:
why do we use a struct for this? is it something to do with needing a pointer to it? why isn't a `*uint64` good enough for this purpose?